test/runnable/b20470.d

```diff
@@ -0,0 +1,97 @@
+// https://issues.dlang.org/show_bug.cgi?id=20470
+
+alias AliasSeq(Args...) = Args;
+
+int g, h;
+
+void test0()
+{
+    static struct S
+    {
+        int a, b;
+        float c = 0, d = 0;
+        alias fields = AliasSeq!(a, b, c, d);
+        alias ints = AliasSeq!(a, b);
+        alias floats = AliasSeq!(c, d);
+        alias reversed = AliasSeq!(d, c, b, a);
+        alias globals = AliasSeq!(g, h);
+
+        alias properties = AliasSeq!(e, f);
+        @property int e() { return a; }
+        @property void e(int i) { a = i; }
+        @property float f() { return c; }
+        @property void f(float j) { c = j; }
+    }
+
+    S s;
+    assert(s.fields == AliasSeq!(0, 0, 0, 0));
+    s.ints = AliasSeq!(1, 2);
+    assert(s.fields == AliasSeq!(1, 2, 0, 0));
+    s.floats = AliasSeq!(3, 4);
+    assert(s.fields == AliasSeq!(1, 2, 3, 4));
+
+    int a, b;
+    float c, d;
+    AliasSeq!(d, c, b, a) = s.reversed;
+    assert(AliasSeq!(a, b, c, d) == AliasSeq!(1, 2, 3, 4));
+
+    s.globals = AliasSeq!(30, 40);
+    assert(g == 30 && h == 40);
+
+    // Propagating `this` to functions and properties within tuples will be a breaking change.
+    // See `test2()` below for an example of existing code that would need to be fixed.
```

Comment:
This would be worth the breaking change, IMO, but would very hard, if not impossible, to have a deprecation process for it.

@SSoulaimane Could we make it work for this syntax instead:

```D
alias properties = AliasSeq!(this.e, this.f);
```

?